pkg/proxy/access/clusteraccess.go

```diff
@@ -0,0 +1,40 @@
+package access
+
+import (
+	"net/url"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// ClusterAccess holds an information needed to access the namespace in a member cluster for the specific user
```

Comment:
This CA is designed to be used for impersonation only. Let's mention it in the comment.

Reply:
Updated the comment in 967fdab265e038a8df86bf12da73dce210e478bb. Let me know if you would like to add/change anything else.

Reply:
Looks good! Thanks.